port/cpl_vsisimple.cpp

```diff
@@ -1371,20 +1371,57 @@ GIntBig CPLGetPhysicalRAM(void)
         return 0;
     GIntBig nVal = static_cast<GIntBig>(nPhysPages) * nPageSize;
 
-    // In a Docker container the memory might be limited
-    // If no limitation, on 64 bit, 9223372036854771712 is returned.
-    FILE *f = fopen("/sys/fs/cgroup/memory/memory.limit_in_bytes", "rb");
-    if (f)
+#ifdef __linux
+    char szMemoryPath[256];
+    szMemoryPath[0] = 0;
     {
-        char szBuffer[32];
-        const int nRead =
-            static_cast<int>(fread(szBuffer, 1, sizeof(szBuffer) - 1, f));
-        szBuffer[nRead] = 0;
+        FILE *f = fopen("/proc/self/cgroup", "rb");
+        char szLine[256];
+        // Find line like "6:memory:/user.slice/user-1000.slice/user@1000.service"
+        // and store "/user.slice/user-1000.slice/user@1000.service" in szMemoryPath
+        while (fgets(szLine, sizeof(szLine) - 1, f))
```

Comment:
```suggestion
        while (fgets(szLine, sizeof(szLine), f))
```

`fgets` reads one character less than the size passed in, then adds a null.

Reply:
thanks. fixed